packages/core/README.md

```diff
@@ -525,61 +525,61 @@ class MyViolation extends Violation {
+| fn                      | fn: ValidatorFn, type?: string                                   | Function reference as a validator. A short cut for extending Validator class.                                                                        |
+| ignore                  |                                                                  | Converts any input value to undefined.                                                                                                               |
+| any                     |                                                                  | Accepts any value, including undefined and null.                                                                                                     |
+| check                   | ...allOf: Validator[]                                            | Runs all the validators and, if successful, returns the original value discarding any conversions.                                                   |
+| map                     | fn: MappingFn, error?: any                                       | Mapping function to convert a value. Catches and converts errors to Violations                                                                       |
+| optional                | type: Validator, ...allOf: Validator[]                           | Allows null and undefined. For other values, runs first the `type` validator and then `allOf` the rest validators.                                   |
+| required                | type: Validator, ...allOf: Validator[]                           | 1) Requires a non-null and non-undefined value, 2) runs `type` validator and 3) then `allOf` the rest validators.                                    |
+| string                  |                                                                  | Requires string or String.                                                                                                                           |
+| toString                |                                                                  | Converts primitive values to (primitive) strings.                                                                                                    |
+| notNull                 |                                                                  | Requires non-null and non-undefined value.                                                                                                           |
+| nullOrUndefined         |                                                                  | Requires null or undefined value.                                                                                                                    |
+| notEmpty                |                                                                  | Requires non-null, non-undefined and not empty. Works with `string`, `null`, `undefined`, `arrays`, `objects` and anything having a numeric length.  |
```

Comment:
As it already states "non-null, non-undefined and not empty", I think it suffice to say that it works with strings, arrays and objects. 

The "anything having a numeric length" is contradictory in itself: is object with one property length=0 empty or not? I'd simply remove that thing, even though it could be considered a major change, as there are better ways of checking that some property has > 0 value.